File: services/api/src/modules/dev-support/dev-support.controller.ts
import { Body, Controller, Get, Post } from '@nestjs/common';
import { ApiTags } from '@nestjs/swagger';
import { DevSupportService } from './dev-support.service';

@ApiTags('dev')
@Controller('v1/dev')
export class DevSupportController {
  constructor(private readonly devSupportService: DevSupportService) {}

  @Get('status')
  getStatus() {
    return this.devSupportService.getStatus();
  }

  @Post('seed')
  seed(@Body() body?: { reset?: boolean }) {
    return this.devSupportService.seedDemo(Boolean(body?.reset ?? true));
  }
}

File: services/api-java/src/main/java/com/tripin/api/controller/DevSupportController.java
package com.tripin.api.controller;

import com.tripin.api.service.DevSupportService;
import com.tripin.api.web.Requests.DevSeedRequest;
import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1/dev")
public class DevSupportController {
  private final DevSupportService devSupportService;

  public DevSupportController(DevSupportService devSupportService) {
    this.devSupportService = devSupportService;
  }

  @GetMapping("/status")
  public Map<String, Object> getStatus() {
    return devSupportService.getStatus();
  }

  @PostMapping("/seed")
  public Map<String, Object> seed(@RequestBody(required = false) DevSeedRequest request) {
    return devSupportService.seedDemo(request == null || request.reset() == null || request.reset());
  }
}
